### PR TITLE
#2 Ticket ingress GitHub scraper

### PR DESCRIPTION
**Pull request by Sampai28**
### Motivation
Resolves: #2 

### Implemented Changes

- Added `scrape_github_issues.py` to fetch issues from GitHub repos via API
- Added `csv_to_json.py` to process and compress scraped data
- Generated `tickets_final.json.gz` containing issue board metadata and ticket list
- Updated `requirements.txt` with necessary Python dependencies
- Updated README with API key setup and usage instructions
- Updated `.gitignore` to exclude intermediate CSV files and environment variables
- Implements secure API key management using dotenv configuration

## How has this been tested?

- Manually tested scraper against GitHub API with valid personal access token
- Verified CSV to JSON conversion pipeline processes data correctly
- Confirmed final compressed JSON output contains required fields

## Screenshots

N/A

## Checklist:

- [x] I followed code style of the project.
- [x] I've updated all documentation accordingly.
- [ ] I've added tests to cover my changes.
- [x] I've ensured all tests pass (no regressions).

### CODE REVIEW

File: scrape_github_issues.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        issues = repo.get_issues(state='closed', sort='created', direction='desc')
+        
+        count = 0
+        for issue in issues:
+            # Skip pull requests
+            if issue.pull_request:
+                continue
+            
+            # Skip issues with no body
+            if not issue.body or len(issue.body) < 50:
```
Comment (ALearningCurve) [inline scrape_github_issues.py]: We need to keep issues with no body and short bodies. If we filter these out of our training dataset we have a few problems:
 - end up with empty training set for certain repos if they contain insufficiently long body 
 - not a complete representation of the issue board (this hurts generalization)
 
IMO for this staage we shouldn't filter based on body

File: requirements.txt
[the remark this answers is not in the document]
Comment (ALearningCurve) [reply requirements.txt]: This is kind of a more advanced repository setup, but it should make things easier for us in the long run!

To see how I setup uv, tests, and added a justfile to manage commands see my other PR #28

File: scrape_github_issues.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                complexity = 'complex'
+            
+            # Calculate base effort
+            if issue.closed_at and issue.created_at:
+                days_open = (issue.closed_at - issue.created_at).days
+                base_effort = min(days_open * 1.5, 30)  # Cap at 30 hours
+            else:
+                base_effort = {'simple': 3, 'medium': 6, 'complex': 12}[complexity]
+            
+            issues_data.append({
```
[the remark this answers is not in the document]
Comment (Sampai28) [reply scrape_github_issues.py]: I think that could be part of the Feature engineering and data normalization ticket (#4) ? For this ticket, I'm just limiting the attributes to:
- id
- repo
- title
- body
- labels
- assignee
- created_at
- closed_at
- comments_count
- url

Let me know your thoughts on this
